### PR TITLE
feat(rust, python): add streamable udfs

### LINKED ISSUE #6603: Issue while using sink_parquet on a LazyFrame
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

I am getting the below error while using sink_parquet on a LazyFrame. Earlier I was using `.collect()` on the output of the `scan_parquet()` to convert the result into a DataFrame but unfortunately it is not working with larger than RAM datasets. Here is error I received -

`PanicException: sink_parquet not yet supported in standard engine. Use 'collect().write_parquet()'`

I am trying to write the LazyFrame (the output from scan_parquet) into a local file after I added some filter and join conditions on the LazyFrame . 

It seems the error is coming from the below location -

https://github.com/pola-rs/polars/blob/master/py-polars/polars/internals/lazyframe/frame.py#L1235 (In Python)
https://github.com/pola-rs/polars/blob/master/polars/polars-lazy/src/physical_plan/planner/lp.rs#L154 (In Rust) . 

I have tried updating to the latest version 0.15.16 but this issue still exists .

### Reproducible example

```python
pl.scan_parquet("path/to/file1.parquet")
    .with_column(
        pl.col("col2")
        .str.split_exact(by="/", n=2)
        .struct.rename_fields(["level_1", "level_2"])
        .alias("splited_levels")
    )
    .unnest("splited_levels")
    .with_column((pl.col("level_1") + "/" + pl.col("level_2")).alias("col_5"))
    ....followed by additional filters
    .select([pl.exclude(["level_1", "level_2"])])
    .sink_parquet("path/to/result2.parquet")
```

### Expected behavior

The parquet file should be written in local system. Instead I am getting the below error -

`PanicException: sink_parquet not yet supported in standard engine. Use 'collect().write_parquet()'`

### Installed versions

<details>

```
Polars : 0.15.16
```

</details>

### PR DESCRIPTION
**Pull request by ritchie46**
Any function that can operate in batches can now be used in the streaming engine.

```python
(pl.DataFrame({
    "a": [1, 2, 3]
}).lazy()
.map(
    f = lambda df: df.with_columns(pl.col("a").alias("b")),
    schema={"a": pl.Int64, "b": pl.Int64},
    streamable=True,
)).collect(streaming=True)
```

closes #6603

### CODE REVIEW

File: py-polars/tests/unit/test_cfg.py [hunk 1]
```diff
@@ -415,6 +415,7 @@ def test_string_cache() -> None:
     assert_frame_equal(out, expected)
 
 
+@pytest.mark.xdist_group(name="group1")
```
Comment (ritchie46) [inline py-polars/tests/unit/test_cfg.py:418]: @stinodego I marked `"group1"` as the group for any contention. E.g. things depending on global env vars should go to this group.

Comment (stinodego) [reply py-polars/tests/unit/test_cfg.py:418]: I think you can just use monkeypatch for the environment variable, and everything should run fine in parallel? Every test should be able to run in isolation.

Comment (ritchie46) [reply py-polars/tests/unit/test_cfg.py:418]: Moneypatch will not set that variable in rust. Env vars are global.

Comment (stinodego) [reply py-polars/tests/unit/test_cfg.py:418]: Just merge it like this, I'll take a look at this later to see if there's a better way!

**Pull request merged**